app/src/main/java/co/tinode/tindroid/CallFragment.java

```diff
@@ -0,0 +1,786 @@
+            for (RtpSender transceiver : mLocalPeer.getSenders()) {
+                MediaStreamTrack track = transceiver.track();
+                if ((toggleCamera && track instanceof VideoTrack) ||
+                        (!toggleCamera && track instanceof AudioTrack)) {
+                    boolean enabled = !track.enabled();
+                    track.setEnabled(enabled);
+                    b.setImageResource(enabled ? enabledIcon : disabledIcon);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, Bundle savedInstance) {
+        // Check permissions.
+        final MessageActivity activity = (MessageActivity) getActivity();
+        LinkedList<String> missing = UiUtils.getMissingPermissions(activity,
+                new String[]{Manifest.permission.CAMERA, Manifest.permission.RECORD_AUDIO});
+        if (!missing.isEmpty()) {
+            mMediaPermissionLauncher.launch(missing.toArray(new String[]{}));
+            return;
+        }
+        // Got all necessary permissions.
+        this.start();
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View v = inflater.inflate(R.layout.fragment_call, container, false);
+        mLocalVideoView = v.findViewById(R.id.localView);
+        mRemoteVideoView = v.findViewById(R.id.remoteView);
+
+        AudioManager audioManager = (AudioManager)getActivity()
+                .getSystemService(Context.AUDIO_SERVICE);
+        audioManager.setMode(AudioManager.MODE_IN_CALL);
+        audioManager.setSpeakerphoneOn(true);
+
+        // Button click handlers: mute/unmute video/audio, hang up.
+        v.findViewById(R.id.hangupBtn).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                handleCallClose();
+            }
+        });
+        v.findViewById(R.id.toggleCameraBtn).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                toggleMedia((FloatingActionButton)v, true,
+                        R.drawable.ic_outline_videocam_24,
+                        R.drawable.ic_outline_videocam_off_24);
+            }
+        });
+        v.findViewById(R.id.toggleMicBtn).setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                toggleMedia((FloatingActionButton)v, false,
+                        R.drawable.ic_outline_mic_24,
+                        R.drawable.ic_outline_mic_off_24);
+            }
+        });
+        return v;
+    }
+
+    @Override
+    public void onDestroyView() {
+        this.stop();
+        super.onDestroyView();
+    }
+
+    public void start() {
+        // Keep screen on.
+        getActivity().getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
```

Comment:
`getActivity()` may return null.

I usually check `activity != null && !activity.isFinishing() && !activity.isDestroyed()` otherwise it crashes a lot.